libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteLegProgressTest.java

```diff
@@ -17,77 +12,46 @@
 import static junit.framework.Assert.assertTrue;
 
-import java.io.IOException;
-
 public class RouteLegProgressTest extends BaseTest {
 
-  // Fixtures
-  private static final String DIRECTIONS_PRECISION_6 = "directions_v5_precision_6.json";
-
-  private DirectionsRoute route;
-  private RouteLeg firstLeg;
+  @Test
+  public void sanityTest() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
 
-  @Before
-  public void setup() throws IOException {
-    Gson gson = new GsonBuilder()
-      .registerTypeAdapterFactory(DirectionsAdapterFactory.create()).create();
-    String body = loadJsonFixture(DIRECTIONS_PRECISION_6);
-    DirectionsResponse response = gson.fromJson(body, DirectionsResponse.class);
-    route = response.routes().get(0);
-    firstLeg = route.legs().get(0);
+    assertNotNull(routeProgress.currentLegProgress());
   }
 
   @Test
-  public void sanityTest() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(0)
-      .legIndex(0)
-      .build();
-    assertNotNull("should not be null", routeProgress.currentLegProgress());
-  }
+  public void upComingStep_returnsNextStepInLeg() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+
+    routeProgress = routeProgress.toBuilder().stepIndex(5).build();
 
-  @Test
-  public void upComingStep_returnsNextStepInLeg() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(5)
-      .legIndex(0)
-      .build();
     assertTrue(routeProgress.currentLegProgress().upComingStep().geometry()
       .startsWith("so{gfA~}xpgFzOyNnRoOdVqXzLmQbDiGhKqQ|Vie@`X{g@dkAw{B~NcXhPoWlRmXfSeW|U"));
   }
 
   @Test
-  public void upComingStep_returnsNull() {
+  public void upComingStep_returnsNull() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+    DirectionsRoute route = buildTestDirectionsRoute();
+    RouteLeg firstLeg = route.legs().get(0);
+
+
     int lastStepIndex = firstLeg.steps().size() - 1;
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(lastStepIndex)
-      .legIndex(0)
-      .build();
+    routeProgress = routeProgress.toBuilder().stepIndex(lastStepIndex).build();
+
     assertNull(routeProgress.currentLegProgress().upComingStep());
```

Comment:
Should be in the `Arrange` part 👀 `upComingStep_returnsNull` is telling us what's the `Act` here `upComingStep`.
Also 👀 https://github.com/mapbox/mapbox-navigation-android/pull/812#discussion_r182119363

This applies to other modified tests included in this PR (`RouteProgressTest` and `RouteStepProgressTest`).